pkg/language/language.go

```diff
@@ -10,10 +10,16 @@ import (
 	"github.com/wakatime/wakatime-cli/pkg/log"
 )
 
+// Config defines language detection options.
+type Config struct {
+	// GuessLanguage enables detecting lexer language from file contents
+	GuessLanguage bool
+}
+
 // WithDetection initializes and returns a heartbeat handle option, which
 // can be used in a heartbeat processing pipeline to detect and add programming
 // language info to heartbeats of entity type 'file'.
-func WithDetection() heartbeat.HandleOption {
+func WithDetection(params Config) heartbeat.HandleOption {
```

Comment:
It's a `config` here.
```suggestion
func WithDetection(config Config) heartbeat.HandleOption {
```